ccxt_backtesting_exchange/backtester.py

```diff
@@ -86,5 +240,199 @@ def withdraw(self, code: str, amount: float, params={}):
+        if since is not None:
+            orders = orders[orders["timestamp"] >= since]
+
+        for column, value in params.items():
+            if column == "until":
+                orders = orders[orders["timestamp"] < params["until"]]
+
+            elif column == "id":
+                orders = orders[orders.index == params["id"]]
+            elif column in orders.columns:
+                orders = orders[orders[column] == value]
+            else:
+                raise ValueError(f"Invalid column '{column}' in params.")
+
+        # Sort orders by timestamp
+        orders = orders.sort_values(by="timestamp", ascending=False)
+
+        # Limit the number of orders if limit is provided
+        if limit is not None:
+            orders = orders.head(limit)
+
+        # Convert DataFrame to list of dictionaries
+        orders_list = (
+            orders.reset_index()
+            .rename(columns={"index": "id"})
+            .to_dict(orient="records")
+        )
+        return orders_list
+
+    def fetch_order(self, id: str, symbol: str = None, params: dict = {}):
+        """
+        Fetches a single order by its ID.
+
+        :param id: The ID of the order.
+        :param symbol: The trading pair symbol (optional).
+        :param params: Additional parameters specific to the exchange API (optional).
+        :return: The order as a dictionary.
+        """
+        order = self.fetch_orders(symbol=symbol, params={"id": id, **params})
+        if not order or len(order) == 0:
+            raise OrderNotFound(f"Order with id '{id}' not found.")
+        if len(order) > 1:
+            raise OrderNotFound(f"Multiple orders found with id '{id}'.")
+
+        return order[0]
+
+    def fetch_open_orders(self, symbol: str, since=None, limit=None, params: dict = {}):
+        """
+        Fetches open orders for a given symbol.
+
+        :param symbol: The trading pair symbol (e.g., 'BTC/USDT').
+        :param since: Timestamp in milliseconds to fetch orders since.
+        :param limit: The maximum number of orders to return.
+        :param params: Additional parameters specific to the exchange API.
+        :return: A list of open orders.
+        """
+        return self.fetch_orders(
+            symbol, since, limit, {"status": OrderStatus.OPEN.value, **params}
+        )
+
+    def fetch_closed_orders(
+        self, symbol=None, since=None, limit=None, params: dict = {}
+    ):
+        """
+        Fetches closed orders for a given symbol.
+
+        :param symbol: The trading pair symbol (e.g., 'BTC/USDT').
+        :param since: Timestamp in milliseconds to fetch orders since.
+        :param limit: The maximum number of orders to return.
+        :param params: Additional parameters specific to the exchange API.
+        :return: A list of closed orders.
+        """
+        return self.fetch_orders(
+            symbol, since, limit, {"status": OrderStatus.FILLED.value, **params}
```

Comment:
@amustapha I think closed orders are basically any order that is not waiting to be filled. So I believe cancelled order are also closed based on the history in binance, what do you think?

Reply:
nope.

if closed orders only returned filled orders.
There's [fetchCancelledOrders](https://docs.ccxt.com/#/baseSpec?id=fetchcanceledorders) which returns only orders that have been cancelled.
And there's [fetchCancelledAndClosedOrders](https://docs.ccxt.com/#/baseSpec?id=fetchcanceledandclosedorders) which would return both cancelled and filled orders.

Basically, fetchClosedOrders would return same as fetchMyTrades, except that your fetchMyTrades could be more. i.e, trades return executions, and one order can potentially have multiple executions, e.g you can create an order to buy 10 sol, and it could have 4 trades for the order to be filled. On a standard exchange, that's why there's PARTIALLY_FILLED order status, i.e, some of the order quantity has been executed but not all.

Reply:
I get!